pkg/services/repo_service.go

```diff
@@ -66,3 +72,76 @@ func (a *argoCDService) GetApps(ctx context.Context, repoURL string, revision st
 
 	return res, nil
 }
+
+func (a *argoCDService) checkoutRepo(gitClient git.Client, revision string) error {
```

Comment:
I am not sure how best to write unit tests for these. We are not doing anything extra on our side, but just call the ArgoCD `git` util, which is unit tested.

Reply:
I don't think we should aim to 100% coverage,

why does the function get's a git client and not creating one based on the repo?

Reply:
I was hoping to mock the `gitClient` for testing. But, just sending the `repo` object makes more sense.

Reply:
`gitClient` is actually tied to a `repo` instance, and it's not a generic `git` client like the name suggests. I sent `gitClient` with the hope of mocking it out for testing. But, there's not much to test here anyways, as the `git` client is already tested in ArgoCD codebase. I agree with you that sending `repo` instance makes more sense.

Reply:
@OmerKahani I gave it a lot of thought to pass `repo` object to `checkoutRepo` func, but there is a use case where `gitClient` is used later, e.g., https://github.com/argoproj-labs/applicationset/pull/45/files#diff-004c4dc22d5fb77bff9d0d6fd29d999d0e068fc99b996bf91a75213585a040efR93. It did not make sense to initialize gitClient multiple times.

So, what I did here: https://github.com/argoproj-labs/applicationset/pull/45/commits/e4e1e5d57f132790d9b9faef2fd8ae510a21cd14 is rename `gitClient` to `getRepoClient` which explicitly states the role of the variable. The git client instance is tied to a repo.

Reply:
OK, sounds good